### PR TITLE
Determine hub id from service options and use when generating/validating hub keys

### PR DESCRIPTION
**Pull request by tconway**
http://bamboo.digicat.io/browse/CHCL-MPT202-5

### CODE REVIEW

File: bass/hubkey.py [hunk 1]
```diff
@@ -44,7 +46,7 @@
 PARTS = OrderedDict([
     ('resolver_id', RESOLVER_ID),
     ('schema_version', SCHEMA),
-    ('hub_id', _PATH_PART),  # word char or -
+    ('hub_id', options.hub_id),
```
Comment (louisnayegon) [inline bass/hubkey.py]: Do we use _PATH_PART anymore?

File: requirements.txt [hunk 2]
```diff
@@ -5,3 +5,5 @@ pytest==2.5.2
 pytest-cov==1.7.0
 sphinx==1.2.3
 six==1.9.0
+tornado==4.2.1
```
Comment (louisnayegon) [inline requirements.txt]: Should we implement make egg_locks for this to avoid build issues in the future?

File: bass/hubkey.py (hunk 1 above)
Comment (tconway) [inline bass/hubkey.py]: Its still used for the s0 hubkeys

File: requirements.txt (hunk 2 above)
Comment (tconway) [inline requirements.txt]: Probably, i'll add it

**Comment by tconway** [comment]
New bamboo job: http://bamboo.digicat.io/browse/CHCL-MPT202-6

**Pull request merged**